DQM/L1TMonitor/interface/L1TStage2CaloLayer2.h

```diff
@@ -0,0 +1,131 @@
+#ifndef DQM_L1TMonitor_L1TStage2CaloLayer2_h
+#define DQM_L1TMonitor_L1TStage2CaloLayer2_h
+
+// system include files
+#include <iosfwd>
+#include <memory>
+#include <vector>
+#include <string>
+#include <algorithm>
+
+// user include files
+//   base classes
+#include "FWCore/Framework/interface/EDAnalyzer.h"
+
+//
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/Run.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "DataFormats/Common/interface/OrphanHandle.h"
+
+// stage2 collection
+
+#include "DataFormats/L1Trigger/interface/BXVector.h"
+#include "DataFormats/L1Trigger/interface/CaloSpare.h"
+#include "DataFormats/L1Trigger/interface/EGamma.h"
+#include "DataFormats/L1Trigger/interface/Jet.h"
+#include "DataFormats/L1Trigger/interface/EtSum.h"
+#include "DataFormats/L1Trigger/interface/Tau.h"
+#include "DataFormats/L1Trigger/interface/Muon.h"
+
+// DQM
+#include "DQMServices/Core/interface/DQMStore.h"
+#include "DQMServices/Core/interface/MonitorElement.h"
+#include "DQMServices/Core/interface/DQMEDAnalyzer.h"
+
+
+class L1TStage2CaloLayer2 : public DQMEDAnalyzer {
+  
+ public:
+  
+  L1TStage2CaloLayer2(const edm::ParameterSet & ps);
+
+  virtual ~L1TStage2CaloLayer2();
+
+ protected:
+
+  void analyze(const edm::Event& e, const edm::EventSetup& c);
+  virtual void dqmBeginRun(const edm::Run&, const edm::EventSetup&);
+  virtual void bookHistograms(DQMStore::IBooker&, const edm::Run&, const edm::EventSetup&) override ;
+  virtual void beginLuminosityBlock(const edm::LuminosityBlock&, const edm::EventSetup&);
+
+ private:
+
+  std::string monitorDir_;
+
+  edm::InputTag stage2CaloLayer2JetSource_;
+  edm::InputTag stage2CaloLayer2EGammaSource_;
+  edm::InputTag stage2CaloLayer2TauSource_;
+  edm::InputTag stage2CaloLayer2EtSumSource_;
+
```

Comment:
@jingyucms - these are not needed as member data are they? (you have the corresponding tokens below)